PollyDemos/Async/AsyncDemo02_WaitAndRetryNTimes.cs

```diff
@@ -1,102 +1,94 @@
-﻿using Polly;
-using System;
-using System.Net.Http;
-using System.Threading;
-using System.Threading.Tasks;
-using PollyDemos.OutputHelpers;
+﻿using PollyDemos.OutputHelpers;
 
 namespace PollyDemos.Async
 {
     /// <summary>
-    /// Demonstrates the WaitAndRetry policy.
-    /// Loops through a series of Http requests, keeping track of each requested
+    /// Demonstrates the Retry strategy with delays between retry attempts.
+    /// Loops through a series of HTTP requests, keeping track of each requested
     /// item and reporting server failures when encountering exceptions.
-    /// 
+    ///
     /// Observations: We now have waits among the retries.
     /// In this case, still not enough wait - or not enough retries - for the underlying system to have recovered.
     /// So we still fail some calls.
     /// </summary>
     public class AsyncDemo02_WaitAndRetryNTimes : AsyncDemo
     {
-        private int totalRequests;
-        private int eventualSuccesses;
-        private int retries;
-        private int eventualFailures;
-
         public override string Description =>
             "Compared to previous demo, this adds waits between the retries. Not always enough wait to ensure success, tho.";
 
         public override async Task ExecuteAsync(CancellationToken cancellationToken, IProgress<DemoProgress> progress)
         {
-            if (progress == null) throw new ArgumentNullException(nameof(progress));
+            ArgumentNullException.ThrowIfNull(progress);
+
+            // Let's call a web API service to make repeated requests to a server.
+            // The service is programmed to fail after 3 requests in 5 seconds.
 
             eventualSuccesses = 0;
             retries = 0;
             eventualFailures = 0;
+            totalRequests = 0;
 
-            progress.Report(ProgressWithMessage(nameof(AsyncDemo02_WaitAndRetryNTimes)));
-            progress.Report(ProgressWithMessage("======"));
-            progress.Report(ProgressWithMessage(string.Empty));
+            PrintHeader(progress, nameof(AsyncDemo02_WaitAndRetryNTimes));
 
-            // Define our policy:
-            var policy = Policy.Handle<Exception>().WaitAndRetryAsync(
-                3, // Retry 3 times
-                attempt => TimeSpan.FromMilliseconds(200), // Wait 200ms between each try.
-                (exception, calculatedWaitDuration) => // Capture some info for logging!
+            // Define our strategy:
+            var strategy = new ResiliencePipelineBuilder().AddRetry(new()
+            {
+                ShouldHandle = new PredicateBuilder().Handle<Exception>(),
+                MaxRetryAttempts = 3,
+                Delay = TimeSpan.FromMilliseconds(200),  // Wait 200ms between each try
```

Comment:
Then we don't need to change two things if we changed the value of the delay. Feels a bit self-descriptive to me in any case.

```suggestion
                Delay = TimeSpan.FromMilliseconds(200),  // Wait between each try
```